components/brave_wallet_ui/components/shared/add-custom-token-form/add-custom-token-form.tsx

```diff
@@ -0,0 +1,348 @@
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenContractAddress(event.target.value)
+  }, [hasError])
+
+  const handleTokenDecimalsChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenDecimals(event.target.value)
+  }, [hasError])
+
+  const handleCoingeckoIDChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setCoingeckoID(event.target.value)
+  }, [hasError])
+
+  const handleIconURLChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setIconURL(event.target.value)
+  }, [hasError])
+
+  // methods
+  const resetInputFields = React.useCallback(() => {
+    setTokenName('')
+    setTokenContractAddress('')
+    setTokenSymbol('')
+    setTokenDecimals('')
+    setTokenID('')
+    setCoingeckoID('')
+    setIconURL('')
+  }, [])
+
+  const onClickAddCustomToken = React.useCallback(() => {
+    if (!customAssetsNetwork) return
+
+    if (foundTokenInfoByContractAddress) {
+      if (foundTokenInfoByContractAddress.isErc721) {
+        let token = foundTokenInfoByContractAddress
+        token.tokenId = tokenID ? new Amount(tokenID).toHex() : ''
+        token.logo = token.logo ? token.logo : iconURL
+        token.chainId = customAssetsNetwork.chainId
+        onAddCustomAsset(token)
+        return
+      }
+      let foundToken = foundTokenInfoByContractAddress
+      foundToken.coingeckoId = coingeckoID !== '' ? coingeckoID : foundTokenInfoByContractAddress.coingeckoId
+      foundToken.logo = foundToken.logo ? foundToken.logo : iconURL
+      foundToken.chainId = customAssetsNetwork.chainId
+      onAddCustomAsset(foundToken)
+    } else {
+      const newToken: BraveWallet.BlockchainToken = {
+        contractAddress: tokenContractAddress,
+        decimals: Number(tokenDecimals),
+        isErc20: customAssetsNetwork.coin !== BraveWallet.CoinType.SOL && !tokenID,
+        isErc721: customAssetsNetwork.coin !== BraveWallet.CoinType.SOL && !!tokenID,
+        name: tokenName,
+        symbol: tokenSymbol,
+        tokenId: tokenID ? new Amount(tokenID).toHex() : '',
+        logo: iconURL,
+        visible: true,
+        coingeckoId: coingeckoID,
+        chainId: customAssetsNetwork.chainId,
+        coin: customAssetsNetwork.coin
+      }
+      onAddCustomAsset(newToken)
+    }
+    onHideForm()
+  }, [tokenContractAddress, foundTokenInfoByContractAddress, customAssetsNetwork, iconURL, tokenDecimals, tokenName, tokenSymbol, tokenID, coingeckoID])
+
+  const onToggleShowAdvancedFields = React.useCallback(() => {
+    setShowAdvancedFields(!showAdvancedFields)
+  }, [showAdvancedFields])
+
+  const onHideNetworkDropDown = React.useCallback(() => {
+    if (showNetworkDropDown) {
+      setShowNetworkDropDown(false)
+    }
+  }, [showNetworkDropDown])
```

Comment:
seems like you can just `setShowNetworkDropDown(false)` without needed to check prior state

Reply:
@josheleonard This is called when a user clicks outside network dropdown, in this case we need to check if it is open before setting it `false`.